app/pages/project/index.cjsx

```diff
@@ -87,8 +88,12 @@ ProjectPage = React.createClass
       @context.geordi?.remember projectToken: nextProps.project?.slug
 
     # Only call to get workflow if we know if there is a user or not and the project is finished loading
-    if nextContext.initialLoadComplete and not nextProps.loading and nextProps.preferences and @state.activeWorkflows.length is 0
-      @getAllWorkflows(nextProps.project)
+    if nextContext.initialLoadComplete and not nextProps.loading and nextProps.preferences and @state.activeWorkflows.length is 0 and not @state.loadingSelectedWorkflow
```

Comment:
Yep, this is the logic that I mentioned needed further refactor in my PR. These checks are to prevent fetching the workflow multiple times while we're waiting for the initial user fetch and the project fetch since it's in `componentWillReceiveProps` part of the lifecycle. What I'd like to do is move the workflow fetch into the set of promises in `fetchProjectData` here: https://github.com/zooniverse/Panoptes-Front-End/blob/master/app/pages/project/index.cjsx#L322

I don't think I would hold up this PR for that though. Extracting it out into its own function is a good step toward that.